### PR TITLE
feat: prepare for sdk v2 updates

### PR DESCRIPTION
**Pull request by xgambitox**
- Refactor token service
- Add subgraph env variables

### CODE REVIEW

**Comment by xgambitox** [comment]
Waiting for sdk release

**Comment by FoxTheSin** [comment]
pre release SDK released

**Comment by FoxTheSin** [comment]
testing Meta Mask (fantom) withdraw & deposit
withdraw fBEETS = good

**Comment by FoxTheSin** [comment]
network switch looks good for MM

**Comment by FoxTheSin** [comment]
token list looks good enough
image

**Comment by FoxTheSin** [comment]
eth simulation pass
image

**Comment by FoxTheSin** [comment]
ledger connection looks good

**Review by FoxTheSin — APPROVED** [review]

File: .env.example [hunk 1]
```diff
@@ -14,3 +14,4 @@ REACT_APP_FORTMATIC_KEY=pk_test_886ADCAB855632AA
 REACT_APP_PORTIS_KEY=b2b7586f-2b1e-4c30-a7fb-c2d1533b153b
 
 REACT_APP_ZAPPER_API_KEY=96e0cc51-a62e-42ca-acee-910ea7d2a241
+REACT_APP_YEARN_SUBGRAPH_KEY=
```
Comment (FiboApe) [inline .env.example:17]: should we add a default one?

File: src/core/frameworks/yearnSdk/index.ts [hunk 2]
```diff
@@ -7,14 +7,20 @@ export class YearnSdkImpl implements YearnSdk {
...
-    const { SUPPORTED_NETWORKS } = config;
+    const { SUPPORTED_NETWORKS, YEARN_SUBGRAPH_ID, YEARN_SUBGRAPH_KEY } = config;
 
     SUPPORTED_NETWORKS.forEach((network) => {
       const providerType = getProviderType(network);
       const provider = web3Provider.getInstanceOf(providerType);
       const networkId = getNetworkId(network) as SdkNetwork;
       const sdkInstance = new Yearn(networkId, {
         provider,
+        ...(YEARN_SUBGRAPH_KEY && {
```
Comment (FiboApe) [inline src/core/frameworks/yearnSdk/index.ts:18]: we should have a ` || {}` at the end of this to send correct parameters?

File: src/core/services/TokenService.spec.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import { createMockToken } from '@src/test';
```
Comment (FiboApe) [inline src/core/services/TokenService.spec.ts:1]: love having tests <3

File: .env.example (hunk 1 above)
Comment (xgambitox) [reply .env.example:17]: Dont have one that I know of, but would be good to add later for devs.

File: src/core/frameworks/yearnSdk/index.ts (hunk 2 above)
Comment (xgambitox) [reply src/core/frameworks/yearnSdk/index.ts:18]: Is one of those js hacky things. Here some response I found https://stackoverflow.com/questions/64603492/i-dont-understand-about-spread-syntax-inside-objects

**Review by turtlemoji — APPROVED** [review]

**Pull request merged**